File: services/Consumers/Core/Domain/Entities/Consumer.cs
using Consumers.Core.Domain.Abstractions.Entities;
using Consumers.Core.Domain.Enums;

namespace Consumers.Core.Domain.Entities;

public class Consumer : Entity
{
    public required string FirstName { get; set; }
    public required string LastName { get; set; }
    public required string Email { get; set; }
    public required string CellPhone { get; set; }
    public required ConsumerType Type { get; set; }
}

File: services/ControlPanel/models/Consumer.ts
import type { Entity } from "./Entity";

export type Consumer = Entity & {
    firstName: string;
    lastName: string;
    email: string;
    cellPhone: string;
    type: ConsumerType;
}

export enum ConsumerType {

    NotDefined = 1,
    Individual = 2,
    Company = 2
}

export const ConsumerTypeOptions = [
    { value: ConsumerType.NotDefined, label: "Μη Ορισμένο" },
    { value: ConsumerType.Individual, label: "Φυσικό Πρόσωπο" },
    { value: ConsumerType.Company, label: "Επιχείρηση" }
];
